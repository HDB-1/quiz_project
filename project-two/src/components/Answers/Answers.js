import React from "react";
import Answer from "../../containers/Answer/Answer";

export default function Answers(props) {
  let answers = [props.correct_answer, ...props.incorrect_answers];

  for (let i = answers.length - 1; i >= 0; i--) {
    let arrayItemForSwitching = answers[i];
    let targetIndex = Math.floor(Math.random() * (answers.length - 1));
    answers[i] = answers[targetIndex];
    answers[targetIndex] = arrayItemForSwitching;
  }

  return (
    <div>
      {answers.map((answer, index) => {
        return <Answer key={index} id={index} answer={answer} submit={props.submit} />;
      })}
    </div>
  );
}
