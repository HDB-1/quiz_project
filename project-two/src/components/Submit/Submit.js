import React from 'react'
import './Submit.css'
export default function Submit(props) {
    return (
        <div className='submit'>
            <button id='skipBtn' onClick={props.skip}>Skip</button>
            <button id='submitBtn' onClick={() => props.submit(props.answerSelected)}>Submit</button>
        </div>
    )
}
