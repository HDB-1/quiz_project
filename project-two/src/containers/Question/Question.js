import React, { Component } from 'react'
import Answers from '../../components/Answers/Answers'
import Submit from '../../components/Submit/Submit'
export class Question extends Component {

    render() {
        return (
            <div>
                <h4 id="questionName">{this.props.questionInfo.question}</h4>
                <Answers answers={["1","2","3","4"]} 
                correct_answer={this.props.questionInfo.correct_answer}
                incorrect_answers={this.props.questionInfo.incorrect_answers}
                />
                <Submit skip={this.props.skip} submit={this.props.submit}/>
            </div>
        )
    }
}
export default Question