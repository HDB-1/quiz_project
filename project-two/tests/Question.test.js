import React from 'react';
import { shallow , mount } from 'enzyme';
import Question from '../src/containers/Question/Question';
import Answers from '../src/components/Answers/Answers'
import Submit from '../src/components/Submit/Submit'

let testInfo =     {
  category: "General Knowledge",
  difficulty: "easy",
  question: "The likeness of which president is featured on the rare $2 bill of USA currency?",
  correct_answer: "Thomas Jefferson",
  incorrect_answers: [
  "Martin Van Buren",
  "Ulysses Grant",
  "John Quincy Adams"
  ]
  }

describe('Shallow Question', () => {
  let wrapper;
  
  beforeEach(() => wrapper = shallow(<Question questionInfo={testInfo}/>));// this automatically tests for rendering without crashing.

  it("Should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render an <Answers/> and <Submit/> component and an element with ID questionName', () => {
      expect(wrapper.find(Answers).length).toEqual(1)
      expect(wrapper.find(Submit).length).toEqual(1)
      expect(wrapper.find("#questionName").length).toEqual(1)
  })
  it('question name component should render {props.questionInfo.question}', () => {
    expect(wrapper.find("#questionName").at(0).text()).toEqual(testInfo.question);
  });

});
describe('Mounted Question', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Question questionInfo={testInfo}/>)
  })

  it('<Answers /> component should render with props correct_answer and incorrect_answers(array)', () => {
    expect(wrapper.find(Answers).props()).toHaveProperty('correct_answer' , testInfo.correct_answer);
    expect(wrapper.find(Answers).props()).toHaveProperty('incorrect_answers' , testInfo.incorrect_answers);
  })
})