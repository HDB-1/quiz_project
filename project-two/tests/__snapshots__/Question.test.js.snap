// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Question Should render correctly 1`] = `
<div>
  <h4
    id="questionName"
  >
    The likeness of which president is featured on the rare $2 bill of USA currency?
  </h4>
  <Answers
    correct_answer="Thomas Jefferson"
    incorrect_answers={
      Array [
        "Martin Van Buren",
        "Ulysses Grant",
        "John Quincy Adams",
      ]
    }
  />
  <Submit />
</div>
`;
